src/main/java/org/colonelkai/publictransit/line/LineDirection.java

```diff
@@ -0,0 +1,45 @@
+package org.colonelkai.publictransit.line;
+
+import org.colonelkai.publictransit.node.Node;
+
+import java.util.List;
+import java.util.Optional;
+
+public enum LineDirection {
+
+    //one way normal
+    POSITIVE_SINGLE(false, 1),
+    //two way normal
+    POSITIVE_FLIP(true, 1),
+    //one way reversed
+    NEGATIVE_SINGLE(false, -1);
+
+    private final int addition;
+    private final boolean canFlipBack;
+
+    LineDirection(boolean canFlipBack, int addition) {
+        this.addition = addition;
+        this.canFlipBack = canFlipBack;
```

Comment:
What does "Can Flip Back" mean?

Reply:
As in not one way, therefore if your at the last stop on the nodes and you request the next, it will flip back. 

Current implementation likely has a bug where the next node after flipping back is the end one again

Reply:
Fair enough, this is good. Also yeah, it definitely will be the end one again, and again...

We need to change the direction to the opposite (to positive if negative, and to negative if positive) after a flipback is done.

Reply:
Got a fix for it, but this is the main reason why i wanted to reverse the list. Got a fix for it, but got 3 classes all depending on each other now